File: VB/Q279518.Web/Northwind.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace Northwind

	<Persistent("Categories")> _
	Public Class Category
		Inherits XPLiteObject
		Private fCategoryID As Integer
		<Key(True)> _
		Public Property CategoryID() As Integer
			Get
				Return fCategoryID
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue(Of Integer)("CategoryID", fCategoryID, value)
			End Set
		End Property
		Private fCategoryName As String
		<Size(15)> _
		Public Property CategoryName() As String
			Get
				Return fCategoryName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("CategoryName", fCategoryName, value)
			End Set
		End Property
		<Association("Category-Products")> _
		Public ReadOnly Property Products() As XPCollection(Of Product)
			Get
				Return GetCollection(Of Product)("Products")
			End Get
		End Property
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Sub New()
			MyBase.New(Session.DefaultSession)
		End Sub
		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
		End Sub
	End Class

	<Persistent("Products")> _
	Public Class Product
		Inherits XPLiteObject
		Private fProductID As Integer
		<Key(True)> _
		Public Property ProductID() As Integer
			Get
				Return fProductID
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue(Of Integer)("ProductID", fProductID, value)
			End Set
		End Property
		Private fProductName As String
		<Size(40)> _
		Public Property ProductName() As String
			Get
				Return fProductName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("ProductName", fProductName, value)
			End Set
		End Property
		Private fSupplier As Supplier
		<Persistent("SupplierID"), Association("Supplier-Products")> _
		Public Property Supplier() As Supplier
			Get
				Return fSupplier
			End Get
			Set(ByVal value As Supplier)
				SetPropertyValue(Of Supplier)("Supplier", fSupplier, value)
			End Set
		End Property
		Private fCategoryID As Category
		<Persistent("CategoryID"), Association("Category-Products")> _
		Public Property Category() As Category
			Get
				Return fCategoryID
			End Get
			Set(ByVal value As Category)
				SetPropertyValue(Of Category)("Category", fCategoryID, value)
			End Set
		End Property
		Private fQuantityPerUnit As String
		<Size(20)> _
		Public Property QuantityPerUnit() As String
			Get
				Return fQuantityPerUnit
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("QuantityPerUnit", fQuantityPerUnit, value)
			End Set
		End Property
		Private fUnitPrice As Decimal
		Public Property UnitPrice() As Decimal
			Get
				Return fUnitPrice
			End Get
			Set(ByVal value As Decimal)
				SetPropertyValue(Of Decimal)("UnitPrice", fUnitPrice, value)
			End Set
		End Property
		Private fDiscontinued As Boolean
		Public Property Discontinued() As Boolean
			Get
				Return fDiscontinued
			End Get
			Set(ByVal value As Boolean)
				SetPropertyValue(Of Boolean)("Discontinued", fDiscontinued, value)
			End Set
		End Property
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Sub New()
			MyBase.New(Session.DefaultSession)
		End Sub
		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
		End Sub
	End Class

	<Persistent("Suppliers")> _
	Public Class Supplier
		Inherits XPLiteObject
		Private fSupplierID As Integer
		<Key(True)> _
		Public Property SupplierID() As Integer
			Get
				Return fSupplierID
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue(Of Integer)("SupplierID", fSupplierID, value)
			End Set
		End Property
		Private fCompanyName As String
		<Size(40)> _
		Public Property CompanyName() As String
			Get
				Return fCompanyName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("CompanyName", fCompanyName, value)
			End Set
		End Property
		Private fContactName As String
		<Size(30)> _
		Public Property ContactName() As String
			Get
				Return fContactName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("ContactName", fContactName, value)
			End Set
		End Property
		Private fContactTitle As String
		<Size(30)> _
		Public Property ContactTitle() As String
			Get
				Return fContactTitle
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("ContactTitle", fContactTitle, value)
			End Set
		End Property
		Private fAddress As String
		<Size(60)> _
		Public Property Address() As String
			Get
				Return fAddress
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Address", fAddress, value)
			End Set
		End Property
		Private fCity As String
		<Size(15)> _
		Public Property City() As String
			Get
				Return fCity
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("City", fCity, value)
			End Set
		End Property
		Private fRegion As String
		<Size(15)> _
		Public Property Region() As String
			Get
				Return fRegion
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Region", fRegion, value)
			End Set
		End Property
		Private fPostalCode As String
		<Size(10)> _
		Public Property PostalCode() As String
			Get
				Return fPostalCode
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("PostalCode", fPostalCode, value)
			End Set
		End Property
		Private fCountry As String
		<Size(15)> _
		Public Property Country() As String
			Get
				Return fCountry
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Country", fCountry, value)
			End Set
		End Property
		Private fPhone As String
		<Size(24)> _
		Public Property Phone() As String
			Get
				Return fPhone
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Phone", fPhone, value)
			End Set
		End Property
		Private fFax As String
		<Size(24)> _
		Public Property Fax() As String
			Get
				Return fFax
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("Fax", fFax, value)
			End Set
		End Property
		Private fHomePage As String
		<Size(1073741823)> _
		Public Property HomePage() As String
			Get
				Return fHomePage
			End Get
			Set(ByVal value As String)
				SetPropertyValue(Of String)("HomePage", fHomePage, value)
			End Set
		End Property
		<Association("Supplier-Products")> _
		Public ReadOnly Property Products() As XPCollection(Of Product)
			Get
				Return GetCollection(Of Product)("Products")
			End Get
		End Property
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Sub New()
			MyBase.New(Session.DefaultSession)
		End Sub
		Public Overrides Sub AfterConstruction()
			MyBase.AfterConstruction()
		End Sub
	End Class

End Namespace

File: CS/Q279518.Web/Northwind.cs
using System;
using DevExpress.Xpo;

namespace Northwind {

    [Persistent("Categories")]
    public class Category :XPLiteObject {
        int fCategoryID;
        [Key(true)]
        public int CategoryID {
            get { return fCategoryID; }
            set { SetPropertyValue<int>("CategoryID", ref fCategoryID, value); }
        }
        string fCategoryName;
        [Size(15)]
        public string CategoryName {
            get { return fCategoryName; }
            set { SetPropertyValue<string>("CategoryName", ref fCategoryName, value); }
        }
        [Association("Category-Products")]
        public XPCollection<Product> Products { get { return GetCollection<Product>("Products"); } }
        public Category(Session session) : base(session) { }
        public Category() : base(Session.DefaultSession) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }

    [Persistent("Products")]
    public class Product :XPLiteObject {
        int fProductID;
        [Key(true)]
        public int ProductID {
            get { return fProductID; }
            set { SetPropertyValue<int>("ProductID", ref fProductID, value); }
        }
        string fProductName;
        [Size(40)]
        public string ProductName {
            get { return fProductName; }
            set { SetPropertyValue<string>("ProductName", ref fProductName, value); }
        }
        Supplier fSupplier;
        [Persistent("SupplierID"), Association("Supplier-Products")]
        public Supplier Supplier {
            get { return fSupplier; }
            set { SetPropertyValue<Supplier>("Supplier", ref fSupplier, value); }
        }
        Category fCategoryID;
        [Persistent("CategoryID"), Association("Category-Products")]
        public Category Category {
            get { return fCategoryID; }
            set { SetPropertyValue<Category>("Category", ref fCategoryID, value); }
        }
        string fQuantityPerUnit;
        [Size(20)]
        public string QuantityPerUnit {
            get { return fQuantityPerUnit; }
            set { SetPropertyValue<string>("QuantityPerUnit", ref fQuantityPerUnit, value); }
        }
        decimal fUnitPrice;
        public decimal UnitPrice {
            get { return fUnitPrice; }
            set { SetPropertyValue<decimal>("UnitPrice", ref fUnitPrice, value); }
        }
        bool fDiscontinued;
        public bool Discontinued {
            get { return fDiscontinued; }
            set { SetPropertyValue<bool>("Discontinued", ref fDiscontinued, value); }
        }
        public Product(Session session) : base(session) { }
        public Product() : base(Session.DefaultSession) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }

    [Persistent("Suppliers")]
    public class Supplier :XPLiteObject {
        int fSupplierID;
        [Key(true)]
        public int SupplierID {
            get { return fSupplierID; }
            set { SetPropertyValue<int>("SupplierID", ref fSupplierID, value); }
        }
        string fCompanyName;
        [Size(40)]
        public string CompanyName {
            get { return fCompanyName; }
            set { SetPropertyValue<string>("CompanyName", ref fCompanyName, value); }
        }
        string fContactName;
        [Size(30)]
        public string ContactName {
            get { return fContactName; }
            set { SetPropertyValue<string>("ContactName", ref fContactName, value); }
        }
        string fContactTitle;
        [Size(30)]
        public string ContactTitle {
            get { return fContactTitle; }
            set { SetPropertyValue<string>("ContactTitle", ref fContactTitle, value); }
        }
        string fAddress;
        [Size(60)]
        public string Address {
            get { return fAddress; }
            set { SetPropertyValue<string>("Address", ref fAddress, value); }
        }
        string fCity;
        [Size(15)]
        public string City {
            get { return fCity; }
            set { SetPropertyValue<string>("City", ref fCity, value); }
        }
        string fRegion;
        [Size(15)]
        public string Region {
            get { return fRegion; }
            set { SetPropertyValue<string>("Region", ref fRegion, value); }
        }
        string fPostalCode;
        [Size(10)]
        public string PostalCode {
            get { return fPostalCode; }
            set { SetPropertyValue<string>("PostalCode", ref fPostalCode, value); }
        }
        string fCountry;
        [Size(15)]
        public string Country {
            get { return fCountry; }
            set { SetPropertyValue<string>("Country", ref fCountry, value); }
        }
        string fPhone;
        [Size(24)]
        public string Phone {
            get { return fPhone; }
            set { SetPropertyValue<string>("Phone", ref fPhone, value); }
        }
        string fFax;
        [Size(24)]
        public string Fax {
            get { return fFax; }
            set { SetPropertyValue<string>("Fax", ref fFax, value); }
        }
        string fHomePage;
        [Size(1073741823)]
        public string HomePage {
            get { return fHomePage; }
            set { SetPropertyValue<string>("HomePage", ref fHomePage, value); }
        }
        [Association("Supplier-Products")]
        public XPCollection<Product> Products { get { return GetCollection<Product>("Products"); } }
        public Supplier(Session session) : base(session) { }
        public Supplier() : base(Session.DefaultSession) { }
        public override void AfterConstruction() { base.AfterConstruction(); }
    }

}
